test/test_fermion.py

```diff
@@ -216,6 +217,53 @@ def test_diagonalize_fermionic_hamiltonian_max_dim(self):
         self.assertEqual(sci_dim_b, 10)
         self.assertAlmostEqual(result.sci_state.spin_square(), expected_spin_square)
 
+    def test_diagonalize_fermionic_hamiltonian_empty_string(self):
+        """Test diagonalize_fermionic_hamiltonian when no valid bitstrings for subsampling."""
+        # Build N2 molecule
+        mol = pyscf.gto.Mole()
+        mol.build(
+            atom=[["N", (0, 0, 0)], ["N", (1.0, 0, 0)]],
+            basis="sto-6g",
+            symmetry="Dooh",
+        )
+
+        # Define active space
+        n_frozen = 2
+        active_space = range(n_frozen, mol.nao_nr())
+
+        # Get molecular integrals
+        scf = pyscf.scf.RHF(mol).run()
+        norb = len(active_space)
+        n_electrons = int(sum(scf.mo_occ[active_space]))
+        n_alpha = (n_electrons + mol.spin) // 2
+        n_beta = (n_electrons - mol.spin) // 2
+        nelec = (n_alpha, n_beta)
+        cas = pyscf.mcscf.CASCI(scf, norb, nelec)
+        mo = cas.sort_mo(active_space, base=0)
+        hcore, nuclear_repulsion_energy = cas.get_h1cas(mo)
+        eri = pyscf.ao2mo.restore(1, cas.get_h2cas(mo), norb)
+
+        # Generate invalid samples from ground state
+        strings = ["00" * norb for _ in range(100)]
+
+        bit_array = BitArray.from_samples(strings, num_bits=2 * norb)
+
+        # Diagonalize
+        with pytest.raises(ValueError) as e_info:
+            _ = diagonalize_fermionic_hamiltonian(
```

Comment:
Let's also test that an error is _not_ raised if you pass a value for `initial_occupancies`.